### PR TITLE
[CIP-20] Unit Testing Reward Query

### PR DESCRIPTION
**Pull request by bh2smith**
Based on #199 

Writing code to execute reward extraction queries on both DBs. This introduces some local setup for unit testing the query. In order to avoid lopsided expectation that we can execute the same query on both we introduce an abstraction for having an array of database connections on which we can execute the same query an return the merged results.

### CODE REVIEW

File: tests/queries/test_batch_rewards.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from pandas import DataFrame
+
+from src.pg_client import MultiInstanceDBFetcher
+from tests.db.pg_client import connect_and_populate_db_from
+
+
+class TestBatchRewards(unittest.TestCase):
+    def setUp(self) -> None:
+        db_url = "postgresql+psycopg2://postgres:postgres@localhost:5432/postgres"
+        self.fetcher = MultiInstanceDBFetcher([db_url])
```
Comment (gentrexha) [inline tests/queries/test_batch_rewards.py:13]: I assume this will be extended once the database is set up?

**Review by gentrexha — APPROVED** [review]
LGTM!

File: tests/queries/test_batch_rewards.py (hunk 1 above)
Comment (bh2smith) [reply tests/queries/test_batch_rewards.py:13]: Yea.. In the real setup we would provide both prod and barn urls to this structure. However, for testing we only have one DB.

File: tests/queries/test_batch_rewards.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import os
+import unittest
+
+import pandas.testing
+from dotenv import load_dotenv
+from pandas import DataFrame
+
+from src.pg_client import MultiInstanceDBFetcher
+from tests.db.pg_client import connect_and_populate_db_from
```
Comment (bh2smith) [inline tests/queries/test_batch_rewards.py]:
```suggestion
```

File: tests/queries/test_batch_rewards.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import os
+import unittest
+
+import pandas.testing
+from dotenv import load_dotenv
```
Comment (bh2smith) [inline tests/queries/test_batch_rewards.py]:
```suggestion
```

**Comment by bh2smith** [comment]
Merging this as it is only testing the query #199 that was already approved and merged.

**Pull request merged**